registry/storage/vacuum.go

```diff
@@ -100,3 +120,33 @@ func (v Vacuum) RemoveRepository(repoName string) error {
 
 	return nil
 }
+
+// RemoveArtifactManifest removes a artifact manifest from the filesystem
+// Removes manifest revision file and manifest ref folder if it exists
+func (v Vacuum) RemoveArtifactManifest(name string, artifactDgst digest.Digest) error {
```

Comment:
An artifact manifest is just a manifest with a specific media type. Why special case it here?